src/main/java/org/apache/ibatis/reflection/ParamNameResolver.java

```diff
@@ -48,6 +51,7 @@
   private final SortedMap<Integer, String> names;
 
   private boolean hasParamAnnotation;
+  private boolean useActualParamName;
```

Comment:
Is this variable necessary? No test fails even if I removed it.
If it is necessary for some corner case, there should be a test.

Sorry if I missed something obvious... 😩😪

Reply:
thx! I will confirm my changes again at tomorrow or after.

Reply:
I think it is necessary for keeping  the backward-compatibility when `Configuration#useActualParamName` is `false`. WDYT?

Reply:
Ah, I see. Thank you for the update!
It would be simpler to store the value of `config.isUseActualParamName()` in a final field. i.e.

```java
private final boolean useActualParamName;

public ParamNameResolver(Configuration config, Method method) {
  useActualParamName = config.isUseActualParamName();
  ...
```

Reply:
You are right!  I've applied a your comment via  ce14cec86659e7ea31e846aeb87d3733841a7450.